api/grpcserver/events.go

```diff
@@ -10,25 +10,43 @@ import (
 
 const subscriptionChanBufSize = 1 << 16
 
-func consumeEvents(ctx context.Context, in <-chan interface{}) <-chan interface{} {
-	out := make(chan interface{}, subscriptionChanBufSize)
+// Errors for cases with a full event buffer.
+var (
+	errTxBufferFull          = "tx buffer is full"
+	errLayerBufferFull       = "layer buffer is full"
+	errAccountBufferFull     = "account buffer is full"
+	errRewardsBufferFull     = "rewards buffer is full"
+	errReceiptsBufferFull    = "receipts buffer is full"
+	errActivationsBufferFull = "activations buffer is full"
+	errStatusBufferFull      = "status buffer is full"
+	errErrorsBufferFull      = "errors buffer is full"
+)
+
+func consumeEvents(ctx context.Context, in <-chan interface{}) (out <-chan interface{}, bufFull <-chan struct{}) {
+	outCh := make(chan interface{}, subscriptionChanBufSize)
+	bufFullCh := make(chan struct{})
```

Comment:
i'm wary of unbuffered channels. they block on send until there is a receiver. maybe of size 1?

Reply:
it is used as a notification that we terminated subscription and stream needs to exit, e.g there are not sends over this channel